src/_pytest/cacheprovider.py

```diff
@@ -103,6 +103,22 @@ def set(self, key, value):
             with f:
                 self.trace("cache-write %s: %r" % (key, value))
                 json.dump(value, f, indent=2, sort_keys=True)
+                self._ensure_readme()
+
+    def _ensure_readme(self):
+        content_readme = """# pytest cache directory #
```

Comment:
Please use `textwrap.dedent` so you don't need to have the lines spill over the right side:

```python

    def _ensure_readme(self):
        content_readme = textwrap.dedent("""\
        # pytest cache directory #
 
        This directory contains data from the pytest's cache plugin, \
        which provides the `--lf` and `--ff` options, as well as the `cache` fixture.
         
        **Do not** commit this to version control.
         
        See [the docs](https://docs.pytest.org/en/latest/cache.html) for more information.
        """)
        if self._cachedir.check(dir=True):
            readme_path = self._cachedir.join("README.md")
            if not readme_path.check(file=True):
                readme_path.write(content_readme)
```

Reply:
I knew this was wrong,but even after searching couldn't come up with a solution. Also shouldn't flake point this out?

Reply:
> Also shouldn't flake point this out?

I don't think so, the code is not *wrong*, but using dedent makes it more pleasing to the eyes. :grin